backend/src/api/controllers/auth.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  insertSingleRow,
+  getRows,
+} from '../../helpers/databasefunctions';
+import express from 'express';
+import { QueryResultRow } from 'pg';
+import { NextFunction } from 'connect';
+import { extractPayload, foundData } from '../../utils';
+import { Fields } from 'formidable';
+import { ErrorHandler } from '../../helpers/error';
+
+export const newAccesstoken = async (
+  req: express.Request,
+  res: express.Response,
+  next: NextFunction,
+): Promise<void | never> => {
+  try {
+    const values = (await extractPayload(req)) as Fields;
+    // if we generate our own accesstokens we should validate them somehow.
+    const accesstoken =
+      values[0] !== undefined && typeof values[0] === 'string'
+        ? (values[0] as string)
+        : false;
+    if (accesstoken) {
+      await insertSingleRow(
+        'insert into keys (accesstoken) VALUES ($1) on conflict do nothing',
+        [accesstoken],
+      );
+      const payload: Object = { status: 'success' };
+      res.status(200).send(payload);
+    } else {
+      throw new ErrorHandler(404, { status: 'Validation Error' });
```

Comment:
It is a bit misleading sending a 404. If nothing else is better suited i recommend simply using a 400. 

My opinion is that 404 should be reserved for when a path actually does not exist or the user is not authorized and we want to hide the existence of the path.

Reply:
Ah, that's just a mistake from my part, probably from copying code and not realizing i should change the status.